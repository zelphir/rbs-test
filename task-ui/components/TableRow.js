import React from 'react'
import PropTypes from 'prop-types'

const TableRow = ({ transaction }) => (
  <tr>
    {Object.keys(transaction).map(item => {
      const key = transaction.id + item

      if (item === 'date') {
        const date = new Date(transaction[item])
        return <td key={key}>{date.toLocaleDateString()}</td>
      }

      return <td key={key}>{transaction[item]}</td>
    })}
  </tr>
)

TableRow.propTypes = {
  transaction: PropTypes.shape({
    id: PropTypes.number.isRequired,
    amount: PropTypes.number.isRequired,
    date: PropTypes.string.isRequired,
    business: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    account: PropTypes.string.isRequired
  })
}

export default TableRow
