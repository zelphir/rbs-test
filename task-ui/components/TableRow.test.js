import React from 'react'
import { shallow } from 'enzyme'
import TableRow from './TableRow'

const transaction = {
  id: 1,
  amount: 85.91,
  date: '2012-02-02T00:00:00.000Z',
  business: 'Bode - Kreiger',
  name: 'Savings Account 6670',
  type: 'withdrawal',
  account: '86389677'
}

describe('<TableRow />', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<TableRow transaction={transaction} />)
    expect(wrapper).toMatchSnapshot()
  })
})
