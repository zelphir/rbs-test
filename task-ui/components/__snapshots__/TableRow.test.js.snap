// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRow /> should match the snapshot 1`] = `
<tr>
  <td
    key="1id"
  >
    1
  </td>
  <td
    key="1amount"
  >
    85.91
  </td>
  <td
    key="1date"
  >
    2012-2-2
  </td>
  <td
    key="1business"
  >
    Bode - Kreiger
  </td>
  <td
    key="1name"
  >
    Savings Account 6670
  </td>
  <td
    key="1type"
  >
    withdrawal
  </td>
  <td
    key="1account"
  >
    86389677
  </td>
</tr>
`;
